internal/export/worker.go

```diff
@@ -391,6 +391,12 @@ func (s *Server) retryingCreateIndex(ctx context.Context, eb *model.ExportBatch,
 			}
 			return fmt.Errorf("creating index file for batch %d: %w", eb.BatchID, err)
 		}
+
+		// Mark files that we've previously cared about as expired.
+		if err := s.markExpiredFiles(ctx, eb); err != nil {
```

Comment:
I feel like this call should come before the call to createIndex so that we can mark the files expired, and then create the index.

Reply:
The two sets *shouldn't* overlap, but I think that's kosher. moved.